src/Microsoft.OData.Core/UriParser/Binders/InBinder.cs

```diff
@@ -65,6 +72,8 @@ internal QueryNode BindInOperator(InToken inToken, BindingState state)
                 left = MetadataBindingUtils.ConvertToTypeIfNeeded(left, right.ItemType);
             }
 
+            MetadataBindingUtils.VerifyCollectionNode(right, this.resolver.EnableCaseInsensitive);

```

Comment:
I'm not familiar with this codebase and it does not use nullable context (`#nullable enable`) so I can't say: could `resolver` ever be `null` here?

Reply:
From other similar codes, it seems it should not be null exception the customer explicitly sets as null. .